test/sync/src/androidTest/kotlin/io/realm/test/mongodb/shared/EmailPasswordAuthTests.kt

```diff
@@ -181,55 +193,56 @@ class EmailPasswordAuthTests {
         // Only emails containing realm_tests_do_autoverify will be confirmed
         val email = "do_not_confirm@10gen.com"
         app.setAutomaticConfirmation(false)
+        app.setCustomConfirmation(true)
         try {
             val provider = app.emailPasswordAuth
-            provider.registerUser(email, "123456")
-            app.setCustomConfirmation(true)
-            val exception = assertFailsWith<AppException> {
+            val exception = assertFailsWith<UserNotFoundException> {
                 provider.retryCustomConfirmation(email)
             }
-            assertTrue(exception.message!!.contains("failed to confirm user do_not_confirm@10gen.com"), exception.message)
+            assertTrue(exception.message!!.contains("user not found"), exception.message)
         } finally {
             app.setCustomConfirmation(false)
+            app.setAutomaticConfirmation(true)
         }
     }
 
     @Test
     fun retryCustomConfirmation_noUserThrows() {
-        val email = "test@10gen.com"
+        val email = "realm_pending_${Random.nextULong()}@10gen.com"
         val adminApi = app.asTestApp
         runBlocking {
             adminApi.setAutomaticConfirmation(false)
+            adminApi.setCustomConfirmation(true)
             val provider = app.emailPasswordAuth
             provider.registerUser(email, "123456")
-            adminApi.setCustomConfirmation(true)
             try {
-                provider.retryCustomConfirmation("foo")
+                provider.retryCustomConfirmation("foo@gen.com")
                 fail()
             } catch (error: UserNotFoundException) {
                 assertTrue(error.message!!.contains("user not found"), error.message)
             } finally {
                 adminApi.setCustomConfirmation(false)
+                adminApi.setAutomaticConfirmation(true)
             }
```

Comment:
```suggestion
            assertFailsWith<UserNotFoundException> {
                provider.retryCustomConfirmation("foo@gen.com")
            }.also { error ->
                assertTrue(error.message!!.contains("user not found"), error.message)
            }
            adminApi.setCustomConfirmation(false)
            adminApi.setAutomaticConfirmation(true)
```

Reply:
Actually, this will do the wrong thing if an assertion is thrown, then the state in 224/225 will not be reset correctly which will impact other tests.

Reply:
If an exception is thrown within the `assertFailsWith` block the test won't stop and whatever comes after that will be executed. Am I missing something else?

Reply:
Oh, I completely missed the call to `fail()` - I see it now 🙈